### PR TITLE
[wip] CRM-19520, fixed query for displaying participant if its a free regis…

### PR DESCRIPTION
**Pull request by pradpnayak**
…tration

---
- CRM-19520: Financial Type access control breaks some reports
  https://issues.civicrm.org/jira/browse/CRM-19520

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I don't think that join will perform well on large data sets - but you should be able to resolve the IF when creating the temp table

**Comment by Edzelopez** [comment]
Tested above PR for free registrations with ACL FT enabled on Event Participant report.

**Comment by JoeMurray** [comment]
Edsel, this code needs to be rewritten as suggested to remove IF from join so that it can scale.

**Comment by eileenmcnaughton** [comment]
per comments above, setting to WIP

File: CRM/Report/Form.php [hunk 1]
```diff
@@ -4708,12 +4708,14 @@ public function getPermissionedFTQuery(&$query, $alias = NULL, $return = FALSE)
...
-              GROUP BY {$query->_aliases['civicrm_contribution']}.id";
+              GROUP BY {$query->_aliases['civicrm_contribution']}.id
+              UNION SELECT 0 as contribution_dummy_id
+    ";
     CRM_Core_DAO::executeQuery($sql);
     if (isset($temp)) {
       $query->_aliases['civicrm_line_item'] = $temp;
     }
-    $from = " INNER JOIN civicrm_contribution_temp temp ON {$query->_aliases['civicrm_contribution']}.id = temp.id ";
+    $from = " INNER JOIN civicrm_contribution_temp temp ON IF({$query->_aliases['civicrm_contribution']}.id IS NULL, 1, {$query->_aliases['civicrm_contribution']}.id = temp.id)  ";
```
Comment (Edzelopez) [inline CRM/Report/Form.php]: @JoeMurray Actually, we have approached this differently on instructions from the core team. We are now using the new selectWhereClause hook in an extension which will likely ship with core. Here is the code: https://github.com/JMAConsulting/biz.jmaconsulting.financialaclreport/blob/master/financialaclreport.php

Comment (JoeMurray) [inline CRM/Report/Form.php]: so should this PR be closed or updated?

Comment (Edzelopez) [inline CRM/Report/Form.php]: I think this can be closed

**Pull request closed without merging**